syft/pkg/cataloger/package_exclusions.go

```diff
@@ -0,0 +1,68 @@
+package cataloger
+
+import (
+	"golang.org/x/exp/slices"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/alpm"
+	"github.com/anchore/syft/syft/pkg/cataloger/apkdb"
+	"github.com/anchore/syft/syft/pkg/cataloger/binary"
+	"github.com/anchore/syft/syft/pkg/cataloger/deb"
+	"github.com/anchore/syft/syft/pkg/cataloger/nix"
+)
+
+type CategoryType string
+
+const (
+	OsCatalogerType     CategoryType = "os"
+	BinaryCatalogerType CategoryType = "binary"
+)
+
+var CatalogerTypeIndex = map[CategoryType][]string{
```

Comment:
Nice! Yea that would be a good simplification here.

My only hesitancy to change it back to that is the original config object we had discussed on the issue:
https://github.com/anchore/syft/issues/931#issuecomment-1642641417

I think keeping this as is has two advantages:
1) Is clear to future users/contributors that Os/Binary categorization types were an explicit choice as and additional condition. The parent child designation loses this nuance a little. 
2) It keeps us open to category based configuration options we may want to consider in the future

Reply:
If the concern is that we want to be explicit about OS and binary cataloger types, these could be named
```golang
var osCatalogerTypes = []string { .... }
var binaryCatalogerTypes = []string { .... }
```
> keeps us open to category based configuration options we may want to consider in the future

I'm all for forward-thinking such as being open to more configuration. The suggestion was more that since we're _not_ doing that at the moment, we don't necessarily know what that would look like (although you had an option originally), so it might be better to just make whatever changes at such time as we do change the feature. Again, this is not a blocker and I'll leave it to your discernment.